src/librustc_resolve/diagnostics.rs

```diff
@@ -199,6 +199,21 @@ types:
 http://doc.rust-lang.org/reference.html#types
 "##
 
+E0403: r##"
+Some type parameters have the same name. Example of erroneous code:
+
+```
+fn foo<T, T>(s: T, u: T) {} // error: the name `T` is already used for a type
+                            //        parameter in this type parameter list
+```
+
+Please verify you didn't mispell the type parameters. Example:
```

Comment:
Please verify that none of the type params are misspelled, and rename any clashing parameters.

(missing "that"; passive voice)